internal/model/modelindexer/indexer.go

```diff
@@ -308,6 +309,17 @@ func encodeAny(v interface{}, out *fastjson.Writer) error {
 		return encodeMap(v, out)
 	case map[string]interface{}:
 		return encodeMap(v, out)
+	case float64:
+		// truncate the value with int conversion to
+		// check for decimals.
+		if float64(v) == float64(int64(v)) {
+			// we cannot rely on fastjson.Marshal as it would strip
+			// the leading zero and encode the value as an "int".
+			// So we call AppendFloat directly with precision=1.
+			out.RawBytes(strconv.AppendFloat(nil, float64(v), 'f', 1, 64))
```

Comment:
nit: There's no need to cast `v` to `float64` since the switch/case clause already does that for you.
```suggestion
		if v == float64(int64(v)) {
			// we cannot rely on fastjson.Marshal as it would strip
			// the leading zero and encode the value as an "int".
			// So we call AppendFloat directly with precision=1.
			out.RawBytes(strconv.AppendFloat(nil, v, 'f', 1, 64))
```

Reply:
Good catch!